File: linear_algebra/C++/NVPK_BT8.cpp
#include <iostream>
#include <vector>
#include <iomanip>
using namespace std;

void xuatMaTran(const vector<vector<int>>& maTran, const string& tenMaTran) {
    cout << "\n--- Ma tran " << tenMaTran << " kich thuoc " << maTran.size() << "x" << maTran.size() << " ---" << endl;
    for (const auto& hang : maTran) {
        for (int phanTu : hang) {
            cout << setw(3) << phanTu;
        }
        cout << endl;
    }
}
int main() {
    int n;
    cout << "nhap kich thuoc n(so nguyen duong): ";
    if (!(cin >> n) || n <= 0) {
        cerr << "Error: khong hop le, vui long nhap lai!" << endl;
        return 1; 
    }
    vector<vector<int>> maTranKhong(n, vector<int>(n, 0));

    vector<vector<int>> maTranDonVi(n, vector<int>(n, 0));

    for (int i = 0; i < n; ++i) {
        maTranDonVi[i][i] = 1;
    }
    xuatMaTran(maTranKhong, "Khong");
    xuatMaTran(maTranDonVi, "Don vi");

    return 0;
}

File: linear_algebra/Python/NVPK_BT8.py
def nhap_n():
 while True:
    try:
        n= int(input("Nhập kích thước n cho ma trận(sô nguyên dương): "))
        if n >0:
            return n
        else:
            print("Lỗi: vui lòng nhập số nguyê dương !")
    except ValueError:
        print("Lỗi :phần tử không hợp lệ, vui lòng nhập số nguyên!")
        
def tao_va_xuat_ma_tran():
    n= nhap_n()
    
    ma_tran_khong=[[0 for j in range(n)] for i in range(n)]
    ma_tran_don_vi=[]
    for i in range(n):
         hang = []
         for j in range(n):
             if i == j:
                 hang.append(1)
             else:
                 hang.append(0)
         ma_tran_don_vi.append(hang)
    xuat_ma_tran(ma_tran_khong, "không",n)
    xuat_ma_tran(ma_tran_don_vi, "đơn vị",n)
        
def xuat_ma_tran(ma_tran, ten_ma_tran, kich_thuoc_n):
    print(f"\n-- Ma trận {ten_ma_tran} kích thước {kich_thuoc_n}x{kich_thuoc_n} --")
    for hang in ma_tran:
        print(" ".join(map(str, hang)))
        

if __name__ == "__main__":
    tao_va_xuat_ma_tran()